packages/components/src/Menu/MenuList.tsx

```diff
@@ -97,27 +97,52 @@ export const MenuListInternal = forwardRef(
   ) => {
     const { id, isOpen, setOpen, triggerElement } = useContext(MenuContext)
 
+    /*
+     * track inner focus state to prevent components from clobbering each
+     * other's keyboard listeners when there are multiple Menus on the page
+     */
+    const [menuHasFocus, setMenuHasFocus] = useState(false)
+
     const [renderIconPlaceholder, setRenderIconPlaceholder] = useState(false)
 
-    const innerRef = useRef<null | HTMLElement>(null)
+    const wrapperRef = useRef<null | HTMLDivElement>(null)
+    const focusId = useRef<any>()
 
     const context = {
       compact,
       renderIconPlaceholder,
       setRenderIconPlaceholder,
     }
 
+    useHotkeys('down', () => menuHasFocus && moveFocus(1, 0, wrapperRef), [
+      menuHasFocus,
+    ])
+
+    useHotkeys('up', () => menuHasFocus && moveFocus(-1, -1, wrapperRef), [
+      menuHasFocus,
+    ])
+
+    const handleFocus = () => {
+      /*
+       * delay focus change to prevent state from updating multiple times as
+       * the event bubbles up the dom
+       */
+      clearTimeout(focusId.current)
+      focusId.current = setTimeout(() => setMenuHasFocus(true), 0)
+    }
+
+    const handleBlur = () => {
+      /*
+       * delay focus change to prevent state from updating multiple times as
+       * the event bubbles up the dom
+       */
+      clearTimeout(focusId.current)
+      focusId.current = setTimeout(() => setMenuHasFocus(false), 0)
+    }
```

Comment:
What you have totally works, but you could get away with something simpler too:
```suggestion
    function handleArrow(e: KeyboardEvent, direction: number, initial: number) {
      if (
        wrapperRef.current &&
        wrapperRef.current.contains(document.activeElement)
      ) {
        moveFocus(direction, initial, wrapperRef)
      }
    }

    useHotkeys('down', () => handleArrow(e, 1, 0))
    useHotkeys('up', () => handleArrow(e, -1, -1))
```

Reply:
wow that's such a better solution 😍